### ISSUE TITLE
Unable to send inline reply from notification action button

### ISSUE THREAD

**Issue by bmwertman** [phonegap/phonegap-plugin-push#1446]
### Expected Behaviour
Select the `REPLY` button and have the inline reply field displayed in the notification interface.

### Actual Behaviour
Upon selecting the `REPLY` action button application is brought from the background to the foreground.

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
     Android 5.0.2

#### (Android) What device vendor (e.g. Samsung, HTC, Sony...)
    Oneplus One

#### Cordova CLI version and cordova platform version
    cordova --version                          6.4
    cordova platform version android           6.1
    cordova platform version ios           4.3

#### Plugin version

    cordova plugin version | grep phonegap-plugin-push   1.9.1

#### Sample Push Data Payload
```
{
    priority: "high",
    data: {
            "content-available": "1",
            title: req.body.sender_name,
            body: req.body.message,
            state: req.body.state,
            roomId: req.body.roomId,
            ledColor: [17, 2, 255, 1],
            soundname: "default",
            image: req.body.sender_imgURL,
            sender_imgUrl: req.body.sender_imgURL,
            actions: [
              {
                "icon": "chat",
                "title": "Chat",
                "foreground": true,
                "callback":"showChatter"
              },
              {
                "icon": "reply",
                "title": "Reply",
                "inline": true,
                "foreground": false,
                "callback": "inlineReply"
              }
            ]
          }
  }
```
#### Sample Code that illustrates the problem
app.js inside `$ionicPlatform.ready`
```
  var launchChat = true;
    window.inlineReply = function(){
      launchChat = false;
      console.log("User replied inline");
    }

    window.showChatter = function(){
      console.log("Taking you to the chat");
    }

    // triggered every time notification received
    $rootScope.$on('$cordovaPushV5:notificationReceived',function(event, notification){
      var data = notification.additionalData;
      if(launchChat){
        if(data.sender_imgUrl !== userService.getUser().pic && data.foreground){// Make sure the user isn't getting notified about their own message
          //Notification was received in foreground. Check if the user is already in the room
          if(!$state.is(data.state, { roomId: data.roomId })){
            var queue = document.getElementById('queue');
            if(!queue){ // This is the first queued chatter
              var body = angular.element(document.getElementsByTagName('body')[0]),
                  messageCount = angular.element(document.createElement('span')),
                  imgTag = angular.element(document.createElement('img')),
                  remove = angular.element(document.createElement('div')),
                  iconBackground = angular.element(document.createElement('div')),
                  queue = angular.element(document.createElement('div'));
              imgTag.attr({
                'ng-repeat':'chatter in queuedChatters track by $index',
                src: data.sender_imgUrl,
                'dragula-model':'queuedChatters',
                class: 'chat-notification',
                'ng-click': 'goToChat($event, chatter.state, chatter.roomId)'
              });
              remove.attr({ 
                id:'remove-notification',
                dragula: '"chatter-bag"',
                'dragula-model': 'removeBag'
              });
              iconBackground.attr({ class: 'icon-background icon ion-close-circled' });
              remove.append(iconBackground);
              queue.attr({
                id: 'queue',
                dragula: '"chatter-bag"',
                'dragula-model': 'queueBag'
              });
              queue.append(imgTag);
              body.append(queue);
              body.append(remove);
              $compile(queue)($rootScope);
              $compile(remove)($rootScope);
              $rootScope.removeDrawer = angular.element(document.getElementById('remove-notification'));
            }
            if($rootScope.queuedChatters.length > 0){
              for (var i = $rootScope.queuedChatters.length - 1; i >= 0; i--) {
                value = $rootScope.queuedChatters[i];
                // If it isn't the same person in the same room sending another message, add them
                if(value.roomId !== data.roomId && value.sender_imgUrl !== data.sender_imgUrl && i === 0){
                  $rootScope.queuedChatters.push(data);
                }
              }
            } else {
              $rootScope.queuedChatters.push(data);
            }
            $rootScope.$apply();
          } 
        } else {
          //Notification was received on device tray and tapped by the user.
          $state.go(data.state, {roomId: data.roomId});
          $cordovaPushV5.Push.finish();
        }
      }
    });
```

**Comment by bmwertman** [comment]
I updated the action button JSON to match more closely with what is shown in the [docs](https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/PAYLOAD.md#in-line-replies). Really only switching the order that I add the buttons to the `actions` array, as follows;

```
actions: [
              {
                "icon": "reply",
                "title": "Reply",
                "callback": "inlineReply",
                "foreground": false,
                "inline": true       
              },
              {
                "icon": "chat",
                "title": "Chat",
                "callback":"showChatter"
              }
            ]
          }
```

The `REPLY` button now respects the `"foreground": false` setting and does not bring the app to the foreground or launch it if closed, which it wasn't before. But the notification tray still closes and no inline reply field is being displayed.

**Comment by bmwertman** [comment]
I updated the action button JSON to match more closely with what is shown in the [docs](https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/PAYLOAD.md#in-line-replies). Really only switching the order that I add the buttons to the `actions` array, as follows;

```
actions: [
              {
                "icon": "reply",
                "title": "Reply",
                "callback": "inlineReply",
                "foreground": false,
                "inline": true       
              },
              {
                "icon": "chat",
                "title": "Chat",
                "callback":"showChatter"
              }
            ]
          }
```

The `REPLY` button now respects the `"foreground": false` setting and does not bring the app to the foreground or launch it if closed, which it wasn't before. But the notification tray still closes and no inline reply field is being displayed.

**Comment by macdonst** [comment]
@bmwertman only available in Android N and later https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/PAYLOAD.md#in-line-replies

**Comment by bmwertman** [comment]
@macdonst so if I'm on a device running Android N(7.0) or above, the above code should presumably work. The device I have to test on is Android 5.1. 

What I'm struggling with is that when a user sends a message out I have no way to know whether the user they are messaging is Android N or above or not.

If I presume all users are Android N or above and set `"foreground": false,` for my action button there doesn't seem to be a way to override that in my `"callback": "inlineReply",` inlineReply function in order to do a `$state.go('chatroom', {roomId: 'THIS_CHATROOMS_ID'})`.

What is the standard approach to incorporate this functionality for users who are running Android N or above while still having a graceful fallback that would launch the app to the correct chatroom from a cold start for those that aren't?

**Comment by fredgalvao** [comment]
Make the payload call the same callback, always, which in your case is `inlineReply` afaik. Then, on the callback, you check the information provided by the `device` cordova plugin, and then do something else when it's not Nougat (instead of trying to grab whatever might come from the inline reply functionallity).

**Comment by bmwertman** [comment]
@fredgalvao This is what I had thought as well. 

But when the app is closed, not in the background, and the `inlineReply` is called by my action button in the notification; the app won't launch due to the `"foreground": false` required to do the inline reply, presumably.

How can I tell the app to launch from closed, and override the `"foreground": false`,   if the OS is earlier than Android N?

**Comment by macdonst** [comment]
@bmwertman you are right, if inline reply is true on Android less than 7 it should bring the app to the foreground. Then you can prompt the user for a response.

**Comment by bmwertman** [comment]
So where does that leave this @macdonst? Is there anything I can try on my end to help resolve it?

**Comment by macdonst** [comment]
@bmwertman pull requests are always welcome :)

**Comment by bmwertman** [comment]
I'm happy to work with you on it. I don't have much Java experience. Do you have any hunches about where in the code I might start?

**Comment by macdonst** [comment]
@bmwertman yeah basically it will be this line:

https://github.com/phonegap/phonegap-plugin-push/blob/ca18653d6ff332db41f48824a2d65bd2699ed8bc/src/android/com/adobe/phonegap/push/PushHandlerActivity.java#L52

need to check if `inline` is true and running Android N or higher instead of just seeing if `inline` is true.

**Comment by bmwertman** [comment]
Cool! I'll fork it and see what I can do.

**Comment by bmwertman** [comment]
@macdonst I submitted a [pull request](https://github.com/phonegap/phonegap-plugin-push/pulls) for this.

 It checks for `inline === true` and if Android version is M or earlier. If these conditions are met it sets `foreground` to `true`. 

I wasn't sure if this needed a documentation update. Let me know if there is anything else I can do to make this a better PR. 😄

**Comment by macdonst** [comment]
Bug fix released with 1.9.2
